docs/source/basics.rst

```diff
@@ -3,202 +3,147 @@
+standard gates with the following:
 
 .. code:: python
 
-    p = Program()
-    p.inst(X(0)).measure(0, 0)
-
-.. parsed-literal::
+     from pyquil.gates import *
 
-    <pyquil.quil.Program at 0x101d45a90>
+Let's instantiate a ``Program`` and add an operation to it. We will act an ``X`` gate on qubit 0.
 
+.. code:: python
 
-This program simply applies the :math:`X`-gate to the zeroth qubit,
-measures that qubit, and stores the measurement result in the zeroth
-classical register. We can look at the Quil code that makes up this
-program simply by printing it.
+    p = Program()
+    p += X(0)
 
-.. code:: python
+All qubits begin in the ground state. This means that if we measure a qubit without applying operations on it, we expect to receive
+a 0 result. The ``X`` gate will rotate qubit 0 from the ground state to the excited state, so a measurement immediately
+after should return a 1 result. More details about gate operations are explained in :ref:`intro`.
 
-    print(p)
+We can print our pyQuil program (``print(p)``) to see the equivalent Quil representation:
 
 .. parsed-literal::
 
     X 0
-    MEASURE 0 [0]
 
-Most importantly, of course, we can see what happens if we run this
-program on the Quantum Virtual Machine, or QVM:
+
+This isn't going to be very useful to us without measurements. In pyQuil 2.0, we have to ``DECLARE`` a memory space
+to read measurement results, which we call "readout results" and abbreviate ``ro``. With measurement, our whole program
+looks like this:
 
 .. code:: python
 
-    qvm = get_qc('9q-square-qvm')
-    result = qvm.run_and_measure(p, trials=10)
-    print(result)
+    from pyquil import Program
+    from pyquil.gates import *
 
-Congratulations! You just ran a program on the QVM. The returned value should be:
+    p = Program()
+    ro = p.declare('ro', 'BIT', 1)
+    p += X(0)
+    p += MEASURE(0, ro[0])
 
-.. parsed-literal::
+    print(p)
 
-    [[1]]
+.. parsed-literal::
 
-For more information on what the above result means, and on executing quantum programs on the QVM in
-general, see :ref:`qvm`. Feel free to skip ahead and read about executing programs on the QVM
-(and the QPU for that matter), but don't forget to come back. The remainder of this section of the
-docs will be dedicated to constructing programs in detail, an essential part of becoming fluent in
-quantum programming.
+    DECLARE ro BIT[1]
+    X 0
+    MEASURE 0 ro[0]
 
-Some Program Construction Features
-----------------------------------
+We've instantiated a program, declared a memory space named ``ro`` with one single bit of memory, applied
+an ``X`` gate on qubit 0, and finally measured qubit 0 into the zeroth index of the memory space named ``ro``.
 
-Multiple instructions can be applied at once or chained together. The
-following are all valid programs:
+Awesome! That's all we need to get results back. Now we can actually see what happens if we run this
+program on the QVM. We just have to add a few lines to do this.
```

Comment:
Maybe explicitly writing "... on the Quantum Virtual Machine (QVM)" will help bridge the gap until our users are more familiar with acronyms